pkg/test/framework/api/components/ingress.go

```diff
@@ -15,21 +15,31 @@
 package components
 
 import (
+	"net/url"
 	"testing"
 
+	v1 "k8s.io/api/core/v1"
+
+	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pkg/test/framework/api/component"
 	"istio.io/istio/pkg/test/framework/api/ids"
 )
 
 // Ingress represents a deployed Ingress Gateway instance.
 type Ingress interface {
 	component.Instance
-	// Address returns the external HTTP address of the ingress gateway (or the NodePort address,
-	// when running under Minikube).
-	Address() string
+	// URL returns the external URL of the ingress gateway (or the NodePort address,
+	// when running under Minikube) for the given protocol
+	URL(protocol model.Protocol) (*url.URL, error)
 
 	//  Call makes an HTTP call through ingress, where the URL has the given path.
 	Call(path string) (IngressCallResponse, error)
+
+	// Configure a secret and wait for the existence
+	ConfigureSecretAndWaitForExistence(secret *v1.Secret) (*v1.Secret, error)
```

Comment:
Certainly for things that are mostly init-time for the component this pattern is preferred. It what @ozevren and I have been using.

If there are things the test needs to actuate mid-test then they would need methods.

Reply:
I would like to keep 2 methods to separate the concerns.
The method ```ConfigureSecretAndWaitForExistence``` is used to configure the secrets as described [here](https://istio.io/docs/tasks/traffic-management/secure-ingress/). The secrets are mounted at ```/etc/istio/ingressgateway-certs```
The method ```AddSecretMountPoint``` is a special method, which is only required for our test. It ensures that we are able to mount the egress and ingress certificates in the same path. Otherwise the deployment of some artifacts will fail on the egress-gateway or on the ingress-gateway. This will prevent other artifacts needed for our test to be installed.

Reply:
I would prefer a declarative rather than imperative model here, and use the Configure method to do that, as nathan suggested.

Reply:
See our comment in the conversation above.